addons/point_of_sale/models/pos_session.py

```diff
@@ -649,9 +655,9 @@ def _reconcile_account_move_lines(self, data):
                 lines.reconcile()
 
         # reconcile invoice receivable lines
-        for account_id in order_account_move_receivable_lines:
-            ( order_account_move_receivable_lines[account_id]
-            | invoice_receivable_lines.get(account_id, self.env['account.move.line'])
+        for partner_id, sign, account_id in order_account_move_receivable_lines:
+            ( order_account_move_receivable_lines[(partner_id, sign, account_id)]
+            | invoice_receivable_lines.get((partner_id, sign, account_id), self.env['account.move.line'])
```

Comment:
since you don't use individual values, I think you could just do:

````suggestion
        for key in order_account_move_receivable_lines:
            ( order_account_move_receivable_lines[key]
            | invoice_receivable_lines.get(key, self.env['account.move.line'])
            ).reconcile()
```